src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicer.java

```diff
@@ -320,44 +875,656 @@ protected boolean isInsideWorkingBound(final AtlasEntity entity)
+                                results.get(countryCode).add(result);
+                            });
+                }
+                else if (isSignificantGeometry(clipped))
+                {
+                    CountryBoundaryMap.setGeometryProperty(clipped, ISOCountryTag.KEY, countryCode);
+                    results.get(countryCode).add(clipped);
+                }
+            }
+        }
+        return results;
+    }
+
+    /**
+     * Slice a Line by converting it to 1d geometry, calculating its slices, and creating the new
+     * sliced Lines. If it belongs to just one country or cannot be sliced, update the ISOCountryTag
+     * appropriately instead
+     *
+     * @param line
+     *            The Line to slice as an Area
+     */
+    private void sliceLine(final Line line)
+    {
+        final Time time = Time.now();
+        final LineString jtsLine = JTS_POLYLINE_CONVERTER.convert(line.asPolyLine());
+
+        final Set<org.locationtech.jts.geom.Polygon> intersectingBoundaryPolygons = getIntersectingBoundaryPolygons(
+                jtsLine);
+        if (CountryBoundaryMap.isSameCountry(intersectingBoundaryPolygons))
+        {
+            final String countryCode = CountryBoundaryMap.getGeometryProperty(
+                    intersectingBoundaryPolygons.iterator().next(), ISOCountryTag.KEY);
+            this.stagedLines.get(line.getIdentifier()).withAddedTag(ISOCountryTag.KEY, countryCode);
+            if (!this.isInCountry.test(this.stagedLines.get(line.getIdentifier())))
+            {
+                removeLine(line);
+            }
+            return;
+        }
+
+        // we only want to do this validation if we're going to slice it
+        if (jtsLine.isEmpty() || !jtsLine.isValid())
+        {
+            logger.error("Way {} had invalid geometry, removing from Atlas {}!",
+                    line.getIdentifier(), this.shardOrAtlasName);
+            removeLine(line);
+            return;
+        }
+        final SortedMap<String, Set<LineString>> slices = sliceLineStringGeometry(jtsLine,
+                intersectingBoundaryPolygons);
+        if (slices.isEmpty())
+        {
+            logger.warn("Way {} had empty sliced geometry!", line.getIdentifier());
+            this.stagedLines.get(line.getIdentifier()).withTags(line.getTags())
+                    .withAddedTag(ISOCountryTag.KEY, ISOCountryTag.COUNTRY_MISSING);
+            return;
+        }
+        else if (slices.keySet().size() == 1)
+        {
+            final String countryCode = slices.keySet().iterator().next();
+            this.stagedLines.get(line.getIdentifier()).withAddedTag(ISOCountryTag.KEY, countryCode);
+            if (!this.isInCountry.test(this.stagedLines.get(line.getIdentifier())))
+            {
+                removeLine(line);
+            }
+            return;
+        }
+        else if (slices.values().size() > AbstractIdentifierFactory.IDENTIFIER_SCALE_DEFAULT)
```

Comment:
Similar comment as under sliceArea() about how severe is this error.  Is this case significantly more rare than the similar error for areas, since slicing up one dimensional features should involve many fewer resulting vertices?

Reply:
I don't believe these are common cases for either type of object, but I'll do a bit of inspecting to confirm that!

Reply:
I actually can't find any confirmed cases of this at the moment, but I have seen it in the past-- clearly a rare case, but something we should still check for in the code